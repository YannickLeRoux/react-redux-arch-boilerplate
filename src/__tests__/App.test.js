import Todo from '../components/todo';
import React from 'react'
import { shallow } from 'enzyme';

describe('Todo component should render without any issue', () => {
	let todo,onChange;
	beforeEach(()=>{
		onChange = jest.fn((e)=>(e));
	  	todo = shallow(<Todo id={123} title={'Test Title'} onCheck={onChange} 
          checked={false}/>);
	});
	it('The Todo Component should match the snapshot',()=>{
		expect(todo).toMatchSnapshot();
	});
	it('OnChange handler is called',() => {
		todo.find('input[type="checkbox"]').simulate('change',{ target: { checked: true } });
		expect(onChange).toHaveBeenCalled();
	})
});