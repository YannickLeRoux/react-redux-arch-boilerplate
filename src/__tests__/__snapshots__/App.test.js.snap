// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component should render without any issue The Todo Component should match the snapshot 1`] = `
<div
  className="row justify-content-between border rounded my-3 bg-white p-3"
>
  <h4
    className="col-md-9 "
  >
    Test Title
  </h4>
  <div
    className="col"
  >
    <input
      className="form-control"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
</div>
`;
